### PR TITLE
Splitting TL and NL getvalue

### LINKED ISSUE #125: Nonlinear interpolation interface
Implement NL interpolation (Interpolation TL and NL are conveniently the same), including minor cleanup of the code.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
PR 1 of 3 (or more) on imposing bounds to the analysis (epic #128 ):

- Separation of NL and TL getvalue, which wasn't required when no change of variables were applied.

- Minor code cleanup

closes #125

### CODE REVIEW

File: src/Fields/soca_interpfields_mod.F90 [hunk 1]
```diff
@@ -320,7 +224,110 @@ subroutine interp_tl(fld, locs, vars, geovals, horiz_interp)
...
-  end subroutine interp_tl
+  end subroutine interp
+
+  ! ------------------------------------------------------------------------------
+  !> Apply backward interpolation
+  subroutine getvalues_ad(incr, locs, vars, geovals, traj)
+    type(soca_field),              intent(inout) :: incr
+    type(ufo_locs),                   intent(in) :: locs
+    type(oops_vars),                  intent(in) :: vars
+    type(ufo_geovals),             intent(inout) :: geovals
```
Comment (aerorahul) [inline src/Fields/soca_interpfields_mod.F90]: rename geovals to something else.
L263 refers `geovals%geovals(ivar)`

File: src/Fields/soca_interpfields_mod.F90 [hunk 2]
```diff
@@ -320,7 +224,110 @@ subroutine interp_tl(fld, locs, vars, geovals, horiz_interp)
...
+          incr%ocnsfc%latent_heat(isc:iec,jsc:jec) = incr%ocnsfc%latent_heat(isc:iec,jsc:jec) +&
+               &incr3d(isc:iec,jsc:jec,1)
+
+       case ("upward_sensible_heat_flux_in_air")
+          incr%ocnsfc%sens_heat(isc:iec,jsc:jec) = incr%ocnsfc%sens_heat(isc:iec,jsc:jec) +&
+               &incr3d(isc:iec,jsc:jec,1)
+
+       case ("friction_velocity_over_water")
+          incr%ocnsfc%fric_vel(isc:iec,jsc:jec) = incr%ocnsfc%fric_vel(isc:iec,jsc:jec) +&
+               &incr3d(isc:iec,jsc:jec,1)
```
Comment (aerorahul) [inline src/Fields/soca_interpfields_mod.F90:321]: Please add case default that will abort or trap an error if unrecognized argument.

**Review by aerorahul — CHANGES_REQUESTED** [review]
@guillaumevernieres 1st set of comments. I will continue reviewing.

**Comment by guillaumevernieres** [comment]
@aerorahul & @travissluka : Anychance you could review that pr? If not I will merge by the end of the day.

**Review by travissluka — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]
changes look good.
all tests run as expected.

**Pull request merged**